flink-core/src/main/java/org/apache/flink/core/plugin/PluginLoader.java

```diff
@@ -39,12 +40,12 @@
 	private final ClassLoader pluginClassLoader;
 
 	@VisibleForTesting
-	public PluginLoader(PluginDescriptor pluginDescriptor, ClassLoader parentClassLoader) {
+	public PluginLoader(PluginDescriptor pluginDescriptor, ClassLoader parentClassLoader, String[] alwaysParentFirstPatterns) {
 		this.pluginClassLoader =
 			new ChildFirstClassLoader(
 				pluginDescriptor.getPluginResourceURLs(),
 				parentClassLoader,
-				pluginDescriptor.getLoaderExcludePatterns());
+				ArrayUtils.concat(alwaysParentFirstPatterns, pluginDescriptor.getLoaderExcludePatterns()));
```

Comment:
I think that the intention behind `pluginDescriptor.getLoaderExcludePatterns()` was to pass `alwaysParentFirstPatterns` (and java doc above `PluginDescriptor#loaderExcludePatterns` seems to confirm that), so one of those two is redundant?

Reply:
This is what I have explicitly asked Stefan Richter about, before implementing this,
if the configuration suppose to pass it via `PluginDescriptor.getLoaderExcludePatterns()`.

The letter is designed to allow customization at individual plugin, while here we are passing a system level exclude pattern. So it was fine for him, if we merge both patterns instead of piggybacking on `PluginDescriptor.getLoaderExcludePatterns()`.